tests/Unit/Product/ProductRepositoryTest.php

```diff
@@ -0,0 +1,272 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\Product;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductMetaHandler;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductRepository;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductSyncer;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\ContainerAwareUnitTest;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Tools\HelperTrait\ProductTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Value\ChannelVisibility;
+use Automattic\WooCommerce\GoogleListingsAndAds\Value\MCStatus;
+use WC_Helper_Product;
+use WC_Product;
+
+/**
+ * Class ProductRepositoryTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\Product
+ *
+ * @property ProductRepository  $product_repository
+ * @property ProductMetaHandler $product_meta
+ * @property ProductHelper      $product_helper
+ */
+class ProductRepositoryTest extends ContainerAwareUnitTest {
+
+	use ProductTrait;
+
+	public function test_find_returns_all_supported_products() {
+		// supported
+		$simple_product   = WC_Helper_Product::create_simple_product();
+		$variable_product = WC_Helper_Product::create_variation_product();
+
+		// unsupported
+		WC_Helper_Product::create_external_product();
+
+		$results = $this->product_repository->find();
+
+		$variations = array_map( 'wc_get_product', $variable_product->get_children() );
+		$this->assertCount( count( $variations ) + 2, $results );
+		$this->assertEqualSets(
+			array_merge(
+				[
+					wc_get_product( $simple_product->get_id() ),
+					wc_get_product( $variable_product->get_id() ),
+				],
+				$variations
+			),
+			$results
+		);
```

Comment:
Works fine although I would have refactored it slightly to create something like:
```php
$expected = array_merge(
	[
		wc_get_product( $simple_product->get_id() ),
		wc_get_product( $variable_product->get_id() ),
	],
	array_map( 'wc_get_product', $variable_product->get_children() )
);
```

Then make the assertCount just be a `count( $expected )`. At first glance the variations + 2 isn't clear what you are counting.

Reply:
Changed in https://github.com/woocommerce/google-listings-and-ads/pull/855/commits/cde5c82ce5d8b28d867b19646a72d2724df21649